### PR TITLE
CH1132 | Fix TOC Bug In Xanadu Themes

### PR DESCRIPTION
**Pull request by AndrewGardhouse**
**Title:**
Fix TOC Bug In Xanadu Themes

**Summary:**
> TOC disappears if you scroll down too far, sometimes doesn't come back when you scroll up

**Relevant references:**
[Fix TOC bug in Xanadu themes](https://app.clubhouse.io/xanaduai/story/1132/fix-toc-bug-in-xanadu-themes)
This also fixes [TOC not scrollable until scrolling to bottom of page](https://app.clubhouse.io/xanaduai/story/4397/toc-not-scrollable-until-scrolling-to-bottom-of-page)

**Possible Drawbacks:**
None that I can think of.

**Related GitHub Issues:**
N/A

### CODE REVIEW

**Review by glassnotes — COMMENTED** [review]
@AndrewGardhouse  Nice!! :tada: The TOC is now immediately scrollable and doesn't get eaten at the bottom, it's so great to have that fixed :rocket: 

While playing with it I notice that if you scroll the main page a bit, then try to scroll the TOC again, it doesn't always scroll; it does some of the time though, and seems to do so more reliably when I hover over text in the TOC pane, (Also, scrolling all the way to the top or bottom seems to reset the behaviour back to being fully scrollable.)

File: xanadu_theme/layout.html [hunk 1]
```diff
@@ -256,11 +256,20 @@
...
+          var tocContainer = document.querySelector('.comment-container');
+
+          // Check if the footer is visible
+          if (footerPosition.top < windowHeight && footerPosition.bottom >= 0) {
+              // We want the height of the TOC to be the height of the main content minus how much of the footer is visible.
+              tocContainer.style.height = 'calc(85vh - 45px - ' + (windowHeight - footerPosition.top) + 'px)'
+          } else {
+            // When the user scrolls back to the top of the page after scrolling to the bottom of the page,
+            // We want to reset the TOC container back to it's original height
+            if (tocContainer.style.height !== '85vh') tocContainer.style.height = '85vh';
```
Comment (josh146) [inline xanadu_theme/layout.html:271]: The comments make this so much easier to follow

**Review by josh146 — APPROVED** [review]
@AndrewGardhouse thank you, this is fantastic 🎉 I played around with a few demos, and the TOC scrolling seemed natural and intuitive 💪

**Comment by AndrewGardhouse** [comment]
> While playing with it I notice that if you scroll the main page a bit, then try to scroll the TOC again, it doesn't always scroll; it does some of the time though, and seems to do so more reliably when I hover over text in the TOC pane, (Also, scrolling all the way to the top or bottom seems to reset the behaviour back to being fully scrollable.)

@glassnotes Thank you for reviewing, I wasn't able to reproduce the issue you came across. Could you share a GIF or the URL to the specific page you were having this issue on, please?

**Comment by glassnotes** [comment]
> > While playing with it I notice that if you scroll the main page a bit, then try to scroll the TOC again, it doesn't always scroll; it does some of the time though, and seems to do so more reliably when I hover over text in the TOC pane, (Also, scrolling all the way to the top or bottom seems to reset the behaviour back to being fully scrollable.)
> 
> @glassnotes Thank you for reviewing, I wasn't able to reproduce the issue you came across. Could you share a GIF or the URL to the specific page you were having this issue on, please?

[This](https://3422-214003948-gh.circle-artifacts.com/0/_build/html/demos/learning2learn.html) page works as an example. 

I can't fully pinpoint the conditions, but these steps should help reproduce:
 - At the top of the page, scroll the TOC down to the bottom
 - With the mouse still hovering in the TOC, continue scrolling down - the main demo page will move
 - Start scrolling up, again, still in the TOC - I would expect this to scroll up the TOC since that is where the mouse is, but instead sometimes the main demo page scrolls up
 - Moving the mouse a bit within the TOC then re-enables scrolling up of TOC

**Pull request merged**